Default CXX flags

andresailer opened the pull request:
@gaede @petricm 
I know we are not doing PRs at the moment, but this is the best place to discuss this change without sending emails. I can commit to svn, when we are satisfied.

Here is a macro that would set cxx flags for all packages that depend on ilcutil, as they already all include the ilcsoft_default_settings.

TODO:
- [x] Should set the USE_CXX11 to true somewhere ?
- [ ] Get the compiler from the environment? Avoids the annoying "-D CMAKE_CXX_COMPILER=`which g++`"

andresailer committed:
ilcutil: copy the better solution for CMAKE_INSTALL_PREFIX from lcio

andresailer committed:
Add Macro to set the default cxx flags for all packages depending on ilcutil

ilcsoft_default_settings is already picked up by all other packages so this will set them everywhere

gaede commented:
Looks good. Concerning the open points:  
- would it not make sense to set USE_CXX11='true' in the default settings - and only if overwritten by the user we will not use c++11 ?
- can we also set CMAKE_CXX_COMPILER="which g++" in the default ?
  this has the advantage that the g++ that is in the path (e.g. after initializing the sft gcc4.8) is automatically taken without the need to also set the CXX environment variable ?

petricm commented:
Please don't set CMAKE_CXX_COMPILER=`which g++` but  CMAKE_CXX_COMPILER=`which $DEFAULT_CXX_COMPILER` or something similar.

andresailer commented:
Apparently the setting of CMAKE_CXX_COMPILER needs to happen before project( someProject) happens, so this cannot be done. ( http://stackoverflow.com/a/29904501, https://cmake.org/Wiki/CMake_FAQ#How_do_I_use_a_different_compiler.3F )
We should instead export CC/CXX in the init_ilcsoft.sh

andresailer committed:
Add option for USE_CXX11 and set it to True by default

andresailer committed:
Add missing include of cmake macro

andresailer committed:
Remove ansi for streamlog

andresailer committed:
Correct syntax for option variable, add status about not using cxx11

andresailer committed:
Remove copied cmake macros from streamlog, call the ilcutil cmakemodules instead

andresailer committed:
Export CMAKE_MODULE_PATH to sub projects

gaede commented:
> We should instead export CC/CXX in the init_ilcsoft.sh
> ok, fine with me. We alse set this in the ilcinstall script, so it would be consistent...

andresailer commented:
I made some further changes, now USE_CXX11 is an option that defaults to true and can be disabled when calling cmake. Also some cleanup for streamlog so there are no copies of all the ilcsoft_default... files.

andresailer committed:
FIX: Needs to be FATAL_ERROR, not FATAL.

Cmake will end if c++11 requested but not compatible with compiler
Remove non-existent variable from MESSAGE statement

andresailer commented:
Pushed to svn repository
